packages/framework-core/src/booster-read-models-reader.ts

```diff
@@ -57,12 +61,34 @@ export class BoosterReadModelsReader {
   private async processFetch(readModelRequest: ReadModelRequestEnvelope): Promise<Array<ReadModelInterface>> {
     const readModelMetadata = this.config.readModels[readModelRequest.typeName]
     const searcher = Booster.readModel(readModelMetadata.class)
-    if (readModelRequest.filters) {
-      searcher.filter(readModelRequest.filters)
-    }
+
+    const filters = this.getReadModelFilters(
+      readModelRequest.filters,
+      readModelRequest.currentUser,
+      readModelMetadata.before
+    )
+
+    searcher.filter(filters)
+
     return searcher.search()
   }
 
+  private getReadModelFilters(
+    filters: FilterFor<Class<ReadModelInterface>>,
+    user?: UserEnvelope,
+    beforeHooks?: Array<BeforeFunction>
+  ): FilterFor<ReadModelInterface> {
+    if (beforeHooks) {
+      let currentFilter = filters
+      beforeHooks.map((before: BeforeFunction) => {
+        currentFilter = before(currentFilter, user)
+      })
+      return currentFilter
```

Comment:
I'm 90% sure that this can be done with a reduce function but the one I did it's not working (??)

Reply:
Try replacing this function body with this:
```typescript
return (beforeHooks ?? []).reduce((currentFilter, before) => before(currentFilter, user), filters)
```